## Проблемы/нюансы с заданиями

### В заданиях 5.2, 5.2a информация о выводе показана только на примере

Не хватает уточнения текстом, что вывод сети и маски должен быть упорядочен также, как в примере:

* столбцами
* ширина столбца 10 символов (в двоичном формате
  надо добавить два пробела между столбцами
  для разделения октетов между собой)

### В задании 20.1 обязательно использовать каталог templates как путь в FileSystemLoader, не .

Из-за ошибки в тесте в задании нельзя писать так

```python
env = Environment(loader=FileSystemLoader("."))
```

надо получить каталог из параметра template и использовать его

### Ошибка в тестах 21.1a и 21.3

Чтобы работали тесты, надо исправить файлы теста - удалить 81 строку в файле test_task_21_1a.py и файле test_task_21_3.py.

Было
```python
    correct_return_value = [
        {
            "hostname": "R1_LONDON",
            "uptime": "1 day, 15 hours, 32 minutes",
            "version": "15.3(2)S1",
        }
    ]
```

Надо исправить на:

```python
    correct_return_value = [
        {
            "hostname": "R1_LONDON",
            "version": "15.3(2)S1",
        }
    ]
```

### Ошибка в тесте 22.1d 

Ошибка в тесте 22.1d: в тексте сообщения, которое проверяется в выводе ``"Cоединение с одним из портов существует"``, в самом
начале стоит английская C, из-за этого может получиться отличие при выполнении
задания как на картнке. При этом в задании сообщение, которое надо вывести тоже
написано с англ C, поэтому если скопировать текст сообщения из задания, тест пройдет

![error](https://github.com/pyneng/pyneng.github.io/blob/master/assets/images/error_test_221d.png?raw=true)
